contracts/contracts/poolBooster/PoolBoosterMerkl.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IPoolBooster } from "../interfaces/poolBooster/IPoolBooster.sol";
+import { IMerklDistributor } from "../interfaces/poolBooster/IMerklDistributor.sol";
+
+interface IERC1271 {
+    /**
+     * @dev Should return whether the signature provided is valid for the provided data
+     * @param hash Hash of the data to be signed
+     * @param signature Signature byte array associated with _data
+     */
+    function isValidSignature(bytes32 hash, bytes memory signature)
+        external
+        view
+        returns (bytes4 magicValue);
+}
+
+/**
+ * @title Pool booster for Merkl distributor
+ * @author Origin Protocol Inc
+ */
+contract PoolBoosterMerkl is IPoolBooster, IERC1271 {
+    /// @notice address of merkl distributor
+    IMerklDistributor public immutable merklDistributor;
+    /// @notice address of the OS token
+    IERC20 public immutable rewardToken;
+    /// @notice if balance under this amount the bribe action is skipped
+    uint256 public constant MIN_BRIBE_AMOUNT = 1e10;
+    /// @notice Campaign duration in seconds
+    uint32 public immutable duration; // -> should be immutable
+    /// @notice Campaign type
+    uint32 public immutable campaignType;
+    /// @notice Merkl hash to sign (for signature verification)
+    bytes32 public immutable hashToSign;
+    /// @notice Owner of the campaign
+    address public immutable creator;
+    /// @notice Campaign data
+    bytes public campaignData;
+
+    constructor(
+        address _rewardToken,
+        address _merklDistributor,
+        uint32 _duration,
+        uint32 _campaignType,
+        bytes32 _hashToSign,
+        address _creator,
+        bytes memory _campaignData
+    ) {
+        require(_rewardToken != address(0), "Invalid rewardToken address");
+        require(
+            _merklDistributor != address(0),
+            "Invalid merklDistributor address"
+        );
+        require(_hashToSign != bytes32(0), "Invalid hashToSign address");
+        require(_campaignData.length > 0, "Invalid campaignData");
+        require(_duration > 1 hours, "Invalid duration");
+
+        campaignType = _campaignType;
+        duration = _duration;
+        hashToSign = _hashToSign;
+        creator = _creator;
+
+        merklDistributor = IMerklDistributor(_merklDistributor);
+        rewardToken = IERC20(_rewardToken);
+        campaignData = _campaignData;
+    }
+
+    /// @notice Create a campaign on the Merkl distributor
+    function bribe() external override {
```

Comment:
Since anyone can call `bribe`, do we need to check where the last campaign has finished?
I'm concerned about having overlapping campaigns trying to spend the same reward tokens in the pool booster contract. Or does Merkl prevent overlapping campaigns?

Reply:
Merkl prevent creating campaign with the same Id to prevent overlapping campaigns.
This is how campaignId are generated: 
![image](https://github.com/user-attachments/assets/9d5c5af9-2214-44aa-8569-cd226fd68a7e)

Only startTimestamp and duration will have an impact on the two different call of `MerklePoolBooster::bribe()`. 

So there is two call on the same day, it will revert yes, as Merkl prevent overlapping campaigns .

Reply:
I see. As we calculate the `startTimestamp` from `getTomorrowRoundedTimestamp`, there can only be one bribe per day